snapd.go

```diff
@@ -249,13 +249,11 @@ func action(ctx *cli.Context) {
 	setMaxProcs(cfg.GoMaxProcs)
 
 	c := control.New(cfg.Control)
-
-	coreModules = []coreModule{}
-
-	coreModules = append(coreModules, c)
 	s := scheduler.New(cfg.Scheduler)
+	c.SetTaskScheduler(s)
 	s.SetMetricManager(c)
-	coreModules = append(coreModules, s)
+
+	coreModules = []coreModule{c, s}
```

Comment:
Why does control ever need to talk to the scheduler? Listening for events on the control side makes sense. But control should never need to *know* the scheduler or that one even exists.

That would be a major architecture decision.